sequencer/node/node.go

```diff
@@ -319,275 +316,330 @@ func NewNode(mode Mode, cfg *config.Node, version string) (*Node, error) {
-	if cfg.API.Address != "" {
-		if cfg.Debug.GinDebugMode {
-			gin.SetMode(gin.DebugMode)
-		} else {
-			gin.SetMode(gin.ReleaseMode)
-		}
-		server := gin.Default()
-		server.Use(cors.Default())
-		coord := false
-		var coordnetConfig *api.CoordinatorNetworkConfig
-		if mode == ModeCoordinator {
-			coord = cfg.Coordinator.API.Coordinator
-			if cfg.API.CoordinatorNetwork {
-				// Setup coordinators network configuration
-				// Get libp2p addresses of the registered coordinators
-				// to be used as bootstrap nodes for the p2p network
-
-				//TODO: Check if we'll need this
-				// bootstrapAddrs, err := client.GetCoordinatorsLibP2PAddrs()
-				if err != nil {
-					log.Warn("error getting registered addresses from the SMC or no addresses registered. error:", err.Error())
-				}
-				// Get Ethereum private key of the coordinator
-				keyJSON, err := keyStore.Export(*account, cfg.Coordinator.EthClient.Keystore.Password, cfg.Coordinator.EthClient.Keystore.Password)
-				if err != nil {
-					return nil, common.Wrap(err)
-				}
-				key, err := keystore.DecryptKey(keyJSON, cfg.Coordinator.EthClient.Keystore.Password)
-				if err != nil {
-					return nil, common.Wrap(err)
-				}
-				coordnetConfig = &api.CoordinatorNetworkConfig{
-					BootstrapPeers: nil,
-					EthPrivKey:     key.PrivateKey,
-				}
-			}
+		log.Infow("Found verifier that matches circuit config", "verifierIdx", verifierIdx)
+	} else {
+		verifierIdx = *cfg.Coordinator.Debug.RollupVerifierIndex
+		log.Infow("Using debug verifier index from config", "verifierIdx", verifierIdx)
+		if verifierIdx >= len(scConsts.Rollup.Verifiers) {
+			return nil, common.Wrap(
+				fmt.Errorf("verifierIdx (%v) >= "+
+					"len(scConsts.Rollup.Verifiers) (%v)",
+					verifierIdx, len(scConsts.Rollup.Verifiers)))
 		}
-		var err error
-		nodeAPI, err = NewNodeAPI(cfg.API.Address, cfg.API, api.Config{
-			Version:                  version,
-			ExplorerEndpoints:        cfg.API.Explorer,
-			CoordinatorEndpoints:     coord,
-			Server:                   server,
-			HistoryDB:                historyDB,
-			L2DB:                     l2DB,
-			StateDB:                  stateDB,
-			EthClient:                ethClient,
-			ForgerAddress:            &cfg.Coordinator.ForgerAddress,
-			CoordinatorNetworkConfig: coordnetConfig,
-		}, cfg.API.CoordinatorNetwork, cfg.API.FindPeersCoordinatorNetworkInterval.Duration)
-		if err != nil {
-			return nil, common.Wrap(err)
+		verifier := scConsts.Rollup.Verifiers[verifierIdx]
+		if verifier.MaxTx != cfg.Coordinator.Circuit.MaxTx ||
+			verifier.NLevels != cfg.Coordinator.Circuit.NLevels {
+			return nil, common.Wrap(
+				fmt.Errorf("circuit config and verifier params don't match.  "+
+					"circuit.MaxTx = %v, circuit.NLevels = %v, "+
+					"verifier.MaxTx = %v, verifier.NLevels = %v",
+					cfg.Coordinator.Circuit.MaxTx, cfg.Coordinator.Circuit.NLevels,
+					verifier.MaxTx, verifier.NLevels,
+				))
 		}
 	}
 
+	coord, err = coordinator.NewCoordinator(
```

Comment:
We could change the name from coord to forger.
Also the config for coordinator for started could also be further simplified. Later based on our requirements we can add those part again.